mono/metadata/w32file-unix.c

```diff
@@ -43,6 +37,9 @@
 #ifdef HAVE_DIRENT_H
 # include <dirent.h>
 #endif
+#if __APPLE__
```

Comment:
Shouldn't this be a HAVE_DLFCN_H define?

Reply:
Technically yes, but the assumption was that all Apple platforms have it anyway.